### PR TITLE
Introduce proper formatting on python ,bash and yaml files 

### PR DESCRIPTION
**Pull request by hansinikarunarathne**
# Pull Request Template for Kubeflow manifests Issues

- Please include a summary of changes and the related issue. 
- List any dependencies that are required for this change. 
- Please delete the options that are not relevant.
- The following checklist will help you to satisfy the requirements.

## ✏️ A brief description of the changes

1. > I add [black](https://black.readthedocs.io/en/stable/getting_started.html) GitHub action to format the python files
2. > I used [yamllint](https://yamllint.readthedocs.io/en/stable/quickstart.html) to format the YAML files. Created a GitHub pipeline to automate finding format errors in YAML files. 
3. > I used [shellcheck](https://github.com/koalaman/shellcheck?tab=readme-ov-file#installing-a-pre-compiled-binary) to format the bash files. Created a Github pipeline out of it

## ✅ Contributor checklist
  - [X] All the commits have been _signed-off_  (To pass the `DCO` check)
  - [X] Submit the [Contributor License Agreements](https://cla.developers.google.com/clas) (To pass the `cla/google` check)

---     
 
> You can join the CNCF Slack and access our meetings at the [Kubeflow Community](https://www.kubeflow.org/docs/about/community/) website. Our channel on the CNCF Slack is here [**#kubeflow-platform**](https://app.slack.com/client/T08PSQ7BQ/C073W572LA2).

### CODE REVIEW

**Comment by juliusvonkohout** [comment]
Please rebase to master to fix the conflict with trivy_scan.sh and please only scan files that are changed in pull requests and are in the folders /common and /example for now.

**Comment by juliusvonkohout** [comment]
The error messages should also guide developers on how to fix it in their branch to pass the test.

**Comment by juliusvonkohout** [comment]
I think you can fix the python and bash scripts in this PR.

**Comment by hansinikarunarathne** [comment]
> I think you can fix the python and bash scripts in this PR.

did not understand what you meant?

**Comment by juliusvonkohout** [comment]
> > I think you can fix the python and bash scripts in this PR.
> 
> did not understand what you meant?

There are a lot of warnings about bash and python files. Most of them are maintained directly in this repository. So you can just fix them with the formatting/linting changes and add them here to this PR.

**Comment by hansinikarunarathne** [comment]
> > > I think you can fix the python and bash scripts in this PR.
> > 
> > 
> > did not understand what you meant?
> 
> There are a lot of warnings about bash and python files. Most of them are maintained directly in this repository. So you can just fix them with the formatting/linting changes and add them here to this PR. Everything in /hack should be fixed for example.

Okay I will do that. And Do I need to run this action only in the files change during the PR ?

**Comment by juliusvonkohout** [comment]
Yes we should only check files that are changed in the PR. So PR by PR stuff will be cleaned up.

**Comment by hansinikarunarathne** [comment]
> Yes we should only check files that are changed in the PR. So PR by PR stuff will be cleaned up.
> 
> And I think we should exclude /apps in general.
> 
> Do you have something to check markdown as well?



> Yes we should only check files that are changed in the PR. So PR by PR stuff will be cleaned up.
> 
> And I think we should exclude /apps in general.
> 
> Do you have something to check markdown as well?



> Yes we should only check files that are changed in the PR. So PR by PR stuff will be cleaned up.
> 
> And I think we should exclude /apps in general.
> 
> Do you have something to check markdown as well?

I di the requested changes. After the check, I will add guidelines to find respective errors in the workflow. Can you give me update regarding the state ? @juliusvonkohout

File: apps/kfp-tekton/upstream/base/installs/multi-user/pipelines-profile-controller/run_tests.sh [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#!/bin/sh
```
Comment (juliusvonkohout) [inline apps/kfp-tekton/upstream/base/installs/multi-user/pipelines-profile-controller/run_tests.sh:1]: everything under apps/*/upstream is not maintained by us and shall not be modified.

Comment (hansinikarunarathne) [reply apps/kfp-tekton/upstream/base/installs/multi-user/pipelines-profile-controller/run_tests.sh]: I did the changes as requested

**Comment by juliusvonkohout** [comment]
/lgtm
/approve

we have to test this in practice

**Pull request merged**